R/pkg/R/mllib.R

```diff
@@ -694,8 +694,14 @@ setMethod("predict", signature(object = "KMeansModel"),
 #' }
 #' @note spark.mlp since 2.1.0
 setMethod("spark.mlp", signature(data = "SparkDataFrame"),
-          function(data, blockSize = 128, layers = c(3, 5, 2), solver = "l-bfgs", maxIter = 100,
-                   tol = 0.5, stepSize = 1, seed = 1) {
+          function(data, layers, blockSize = 128, solver = "l-bfgs", maxIter = 100,
+                   tol = 1E-6, stepSize = 0.03, seed = 0x7FFFFFFF) {
+            if (length(layers) <= 1) {
+              stop("layers vector require length > 0.")
+            }
+            if (any(sapply(layers, function(e) as.integer(e) != e))) {
```

Comment:
This way `layers = c(1.0, 2.0)` would pass the `as.integer(e) != e` test.
One possible issue is with how we are handling this on the Scala side. Since we are passing it as `as.array(layers)`, this could end up as double in JVM - would it handle that correctly?

There are other ways to do this but generally coercing to integer is a reasonable approach.

One alternative implementation is this:
```
layers <- as.integer(na.omit(layers))
if (length(layers) <= 1) {
    stop("layers vector must be integer values with length > 0.")
}
```